rpcs3/Emu/Cell/Modules/cellCamera.cpp

```diff
@@ -623,15 +683,29 @@ s32 cellCameraReadEx(s32 dev_num, vm::ptr<CellCameraReadEx> read)
 s32 cellCameraReadComplete(s32 dev_num, u32 bufnum, u32 arg2)
 {
 	cellCamera.todo("cellCameraReadComplete(dev_num=%d, bufnum=%d, arg2=%d)", dev_num, bufnum, arg2);
-	return cellCameraSetAttribute(dev_num, CELL_CAMERA_READFINISH, bufnum, arg2);
+
+	const auto g_camera = fxm::get<camera_thread>();
+
+	if (!g_camera)
+	{
+		return CELL_CAMERA_ERROR_NOT_INIT;
+	}
+
+	const auto status = cellCameraSetAttribute(dev_num, CELL_CAMERA_READFINISH, bufnum, arg2);
+	return status;
```

Comment:
why not return cellCameraSetAttribute(dev_num, CELL_CAMERA_READFINISH, bufnum, arg2);

Reply:
Simple, I didn't like how `return <huge func call>` looked.

Reply:
It just looks funny to me, I think doing it the way MegaMouse has suggested looks best.

Reply:
Fair enough, I'll change it.